[shiftmedia-libgnutls] Add release dll on debug when compile with ngtcp2 and gnutls

### Operating system

windows gitgub action ci

### Compiler

msvc

### Steps to reproduce the behavior

```Shell
./vcpkg install brotli zlib zstd libpsl nghttp2 shiftmedia-libgnutls libssh2 pkgconf gsasl ngtcp2[gnutls] nghttp3

# curl cmake
cmake -DENABLE_DEBUG=ON -DENABLE_UNICODE=OFF -DCURL_USE_SCHANNEL=OFF -DCURL_BROTLI=ON -DCURL_ZSTD=ON -DCURL_USE_LIBPSL=ON -DUSE_NGHTTP2=ON -DBUILD_SHARED_LIBS=OFF -DCURL_USE_LIBSSH2=ON -DCURL_USE_GNUTLS=ON -DCURL_USE_GSASL=ON -DUSE_NGTCP2=ON
```
https://github.com/curl/curl/pull/14495
```


### Failure logs

> @talregev There is an issue with GnuTLS + ngtcp2 which looks vcpkg-related:
>
> When building in Debug mode, curl.exe gets linked with gnutlsd.dll
(the debug GnuTLS DLL), plus ngtcp2.dll. Which in turn pulls in
ngtcp2_crypto_gnutls.dll, which in turn pulls in the release version
of gnutls.dll. This release mode DLL is not copied automatically next to
curl.exe, so it fails to launch.



### Additional context

https://github.com/curl/curl/pull/14495

> adding GnuTLS to the tested packages in `scripts/tests/cmake-user`.

https://github.com/microsoft/vcpkg/pull/40473

> It should report a debug/release mismatch.

Confirmed:

~~~
CMake Error at CMakeLists.txt:81 (message):
  Release lib for 'Debug' build: D:/installed/x64-windows/lib/gnutls.lib
~~~

--- a/ports/shiftmedia-libgnutls/portfile.cmake
+++ b/ports/shiftmedia-libgnutls/portfile.cmake
@@ -173,3 +173,4 @@ configure_file("${SOURCE_PATH}/lib/gnutls.pc.in" "${CURRENT_PACKAGES_DIR}/debug/
 
 vcpkg_fixup_pkgconfig()
 vcpkg_copy_pdbs()
+file(COPY "${CURRENT_PORT_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/gnutls")

--- a/ports/shiftmedia-libgnutls/vcpkg-cmake-wrapper.cmake
+++ b/ports/shiftmedia-libgnutls/vcpkg-cmake-wrapper.cmake
@@ -1,0 +1,2 @@
+find_library(GNUTLS_LIBRARY NAMES gnutls gnutlsd NAMES_PER_DIR)
+_find_package(${ARGS})

--- a/ports/shiftmedia-libgnutls/vcpkg.json
+++ b/ports/shiftmedia-libgnutls/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "shiftmedia-libgnutls",
   "version": "3.8.4",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Unofficial GnuTLS fork with added custom native Visual Studio project build tools. ",
   "homepage": "https://github.com/ShiftMediaProject/gnutls",
   "license": "LGPL-2.1-only",

--- a/scripts/test_ports/cmake-user/vcpkg.json
+++ b/scripts/test_ports/cmake-user/vcpkg.json
@@ -101,6 +101,11 @@
           "platform": "!(uwp & arm)"
         },
         {
+          "$package": "GnuTLS",
+          "name": "libgnutls",
+          "platform": "!windows | mingw"
+        },
+        {
           "$package": "Iconv",
           "$since": "3.11",
           "name": "libiconv"
@@ -146,6 +151,11 @@
         {
           "$package": "PhysFS",
           "name": "physfs"
+        },
+        {
+          "$package": "GnuTLS",
+          "name": "shiftmedia-libgnutls",
+          "platform": "windows & !arm & !mingw & !xbox"
         },
         {
           "$package": "SQLite3",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -8194,7 +8194,7 @@
     },
     "shiftmedia-libgnutls": {
       "baseline": "3.8.4",
-      "port-version": 1
+      "port-version": 2
     },
     "shiftmedia-libgpg-error": {
       "baseline": "1.45",

--- a/versions/s-/shiftmedia-libgnutls.json
+++ b/versions/s-/shiftmedia-libgnutls.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "1d4b38dbf344dc537c0f0e8fae77706d09ef0126",
+      "version": "3.8.4",
+      "port-version": 2
+    },
+    {
       "git-tree": "a784f9e85e153f86f04458fed9d158e76b861eda",
       "version": "3.8.4",
       "port-version": 1

